addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/SqlInjectionScanRuleUnitTest.java

```diff
@@ -344,6 +346,48 @@ void shouldNotAlertIfMultConfirmationExpressionIsNotSuccessfulAndReflectedInResp
         assertThat(alertsRaised, hasSize(0));
     }
 
+    /**
+     * Some checks are based on response content comparison
+     *
+     * <p>But some sites also include the request payload into their response and do some
+     * xml-escaping to be xhtml compatible.
+     *
+     * <p>Many SqlInjection attack payloads contain characters like " and ', which would be escaped
+     * in the response payload (e.g. " becomes &quot; and ' becomes &apos;).
+     *
+     * <p>The SqlInjectionScanRule should exclude these escaped attack payloads before comparing the
+     * response content.
+     */
```

Comment:
the test name usually says what is done, but not why.
I thought that the benefit of this functionality is not obvious to the majority of developers, especially if new to ZAP or security in general (if it was, then it would have been already in place).
So some explaining words why we do this could help to increase the understandings here.
And since I regard (good) tests as a primary source of technical documentation for devs, it was natural to me to combine the answers regarding WHAT and WHY at one common location.

Admittedly, this is very uncommon in the existing ZAP codebase, so if you say that you generally do not want something like this and do not share my thoughts, then I'm totally fine to omit any explaining words.

Reply:
If you think it needs explanation then it should be done where it's actually happening not buried in a test case. Extract a method and comment there, in any case it's obvious why it's being done (look at the other cases already present and the comment of the method, though explicitly mentioning XHTML would definitely not hurt).